src/metaschema/oscal_implementation-common_metaschema.xml

```diff
@@ -429,7 +442,24 @@
                   <field ref="function-performed" min-occurs="1" max-occurs="unbounded">
                         <group-as name="functions-performed" in-json="ARRAY"/>
                   </field>
+                  <field ref="role-id" min-occurs="0" max-occurs="unbounded">
+                        <group-as name="role-ids" in-json="ARRAY"/>
+                  </field>
+                  <field ref="user-uuid" min-occurs="0" max-occurs="unbounded">
+                        <group-as name="user-uuids" in-json="ARRAY"/>
+                  </field>
             </model>
+            <constraint>
+                  <!-- References to roles -->
+                  <index-has-key name="index-metadata-role-id" target=".">
+                        <key-field target="role-id"/>
+                  </index-has-key>
+                  <!-- References to users -->
+                  <index-has-key name="index-system-implementation-user-uuid" target=".">
+                        <key-field target="@uuid"/>
+                  </index-has-key>
+                  <!-- TODO: Add combination constraint requiring authorized-privilege to reference at least one role-id or user-uuid -->
```

Comment:
@david-waltermire what's the syntax for this kind of combination constraint where we'll want the `authorized-privilege` to refence at least one `role-id` or `user-uuid`?

Reply:
I added an expect constraint for this.